TS 5 Regression. tsconfig extends stop resolving references to node_modules

# Bug Report
### 🔎 Search Terms

TS 5, tsconfig, extends, node modules

### 🕗 Version & Regression Information

5.0.2

In latest stable build, 4.9.5, tsconfig extends resolve correctly references to libraries on node modules. However, after trying out 5.0.2 it has stop working


- This changed between versions 4.9.5 and 5.0.2


### ⏯ Playground Link

No Playground, I quickly check and there is no support for tsconfig, package.json,...

### 💻 Code

No code, it's just configuration. Having a package.json declaring

```
 "devDependencies": {
    "@mylibrary/commons": "1.0.0",
    "typescript": "5.0.2"
  }
```

and a tsconfig with:
`"extends": "@mylibrary/commons/configs/tsconfig.backend.json",`

### 🙁 Actual behavior

The command `tsc --build` finish with error:

```
tsconfig.json:2:14 - error TS6053: File '@mylibrary/commons/configs/tsconfig.backend.json' not found.

2   "extends": "@mylibrary/commons/configs/tsconfig.backend.json",

```


### 🙂 Expected behavior

The command `tsc --build` finish correctly.

I just upgraded to TS 5 and found the same issue with one my own node packages ([antlr4-c3 repo](https://github.com/mike-lischke/antlr4-c3, [npm](https://www.npmjs.com/package/antlr4-c3)). With TS 5 I now get the error 

> Cannot find module 'antlr4-c3' or its corresponding type declarations.ts(2307)

Needless to say that I did not change any bit in the project where I'm using the node package. Just upgrading to TS 5 broke the import.

I played a bit with other node module resolution settings, but nothing helped. What sticks out when looking in the bundled node package is that there's no index.js file in the root folder. All the code is in the `out/` folder.

--- a/src/compiler/moduleNameResolver.ts
+++ b/src/compiler/moduleNameResolver.ts
@@ -1689,7 +1689,7 @@ export function nodeModuleNameResolver(moduleName: string, containingFile: strin
 
 /** @internal */
 export function nodeNextJsonConfigResolver(moduleName: string, containingFile: string, host: ModuleResolutionHost): ResolvedModuleWithFailedLookupLocations {
-    return nodeModuleNameResolverWorker(NodeResolutionFeatures.Exports, moduleName, getDirectoryPath(containingFile), { moduleResolution: ModuleResolutionKind.NodeNext }, host, /*cache*/ undefined, Extensions.Json, /*isConfigLookup*/ true, /*redirectedReference*/ undefined);
+    return nodeModuleNameResolverWorker(NodeResolutionFeatures.NodeNextDefault, moduleName, getDirectoryPath(containingFile), { moduleResolution: ModuleResolutionKind.NodeNext }, host, /*cache*/ undefined, Extensions.Json, /*isConfigLookup*/ true, /*redirectedReference*/ undefined);
 }
 
 function nodeModuleNameResolverWorker(features: NodeResolutionFeatures, moduleName: string, containingDirectory: string, compilerOptions: CompilerOptions, host: ModuleResolutionHost, cache: ModuleResolutionCache | undefined, extensions: Extensions, isConfigLookup: boolean, redirectedReference: ResolvedProjectReference | undefined): ResolvedModuleWithFailedLookupLocations {

--- a/tests/baselines/reference/tsconfigExtendsPackageJsonExportsWildcard.errors.txt
+++ b/tests/baselines/reference/tsconfigExtendsPackageJsonExportsWildcard.errors.txt
@@ -1,0 +1,30 @@
+/index.ts(2,1): error TS2454: Variable 'x' is used before being assigned.
+
+
+==== /tsconfig.json (0 errors) ====
+    {
+        "extends": "foo/strict.json"
+    }
+    
+==== /node_modules/foo/package.json (0 errors) ====
+    {
+        "name": "foo",
+        "version": "1.0.0",
+        "exports": {
+            "./*.json": "./configs/*.json"
+        }
+    }
+    
+==== /node_modules/foo/configs/strict.json (0 errors) ====
+    {
+        "compilerOptions": {
+            "strict": true
+        }
+    }
+    
+==== /index.ts (1 errors) ====
+    let x: string;
+    x.toLowerCase();
+    ~
+!!! error TS2454: Variable 'x' is used before being assigned.
+    

--- a/tests/cases/compiler/tsconfigExtendsPackageJsonExportsWildcard.ts
+++ b/tests/cases/compiler/tsconfigExtendsPackageJsonExportsWildcard.ts
@@ -1,0 +1,27 @@
+// @noTypesAndSymbols: true
+// @noEmit: true
+
+// @Filename: /node_modules/foo/package.json
+{
+    "name": "foo",
+    "version": "1.0.0",
+    "exports": {
+        "./*.json": "./configs/*.json"
+    }
+}
+
+// @Filename: /node_modules/foo/configs/strict.json
+{
+    "compilerOptions": {
+        "strict": true
+    }
+}
+
+// @Filename: /tsconfig.json
+{
+    "extends": "foo/strict.json"
+}
+
+// @Filename: /index.ts
+let x: string;
+x.toLowerCase();

